### PR TITLE
[WIP] Caml documentation cleaning

### PR DESCRIPTION
**Pull request by amelieled**
A cleaning on documentation.
I checked that each function has the correct name and the corresponding variables compared to their documentation.
Only the comments have changed.

### CODE REVIEW

File: src/core/sign.ml [hunk 1]
```diff
@@ -48,9 +49,9 @@ let mem : t -> string -> bool =
     making copies of terms when loading an object file. *)
 let loaded : t PathMap.t ref = ref PathMap.empty
 
-(* NOTE that the current module is stored in [loaded] so that the symbols that
-   it contains can be qualified with the name of the module. This behavior was
-   inherited from previous versions of Dedukti. *)
+(** NOTE that the current module is stored in [loaded] so that the symbols
```
Comment (gabrielhdt) [inline src/core/sign.ml:52]: I don't think we want this documentation to be exported

File: src/core/terms.ml [hunk 2]
```diff
@@ -399,7 +399,7 @@ let _Wild : tbox = Bindlib.box Wild
 let _TRef : term option ref -> tbox = fun r ->
   Bindlib.box (TRef(r))
 
-(** [_LLet t a u] lifts let binding [let x := t : a in u<x>]. *)
+(** [_LLet t a u] lifts let binding [let x := t : a in u<x>]. - @PROBLEM ? *)
```
Comment (gabrielhdt) [inline src/core/terms.ml:402]: `@problem`

File: src/core/tree.ml [hunk 3]
```diff
@@ -335,12 +335,13 @@ module CM = struct
...
 
-  (** [get_col n m] retrieves column [n] of matrix [m]. *)
+  (** [get_col ind m] retrieves column [ind] of matrix [m]. *)
   let get_col : int -> t -> term list = fun ind m ->
     List.map (fun {c_lhs ; _} -> c_lhs.(ind)) m.clauses
 
-  (** [score c] returns the score heuristic for column [c].  This score is the
-      number of tree constructors divided by the number of conditions. *)
+  (** [score ts] returns the score heuristic for column [ts].
+      This score is the number of tree constructors divided
```
Comment (gabrielhdt) [inline src/core/tree.ml:343]: Line break

File: src/core/tree.ml [hunk 4]
```diff
@@ -352,21 +353,22 @@ module CM = struct
     let nc, ns = loop (0, 0) ts in
     float_of_int nc /. (float_of_int ns)
 
-  (** [pick_best_among m c] returns the index of the best column of matrix [m]
-      among columns [c]. Here, we mean "best" in the sense of {!val:score}. *)
+  (** [pick_best_among mat columns] returns the index of the best
+      column of matrix [mat] among columns [columns].
+      Here, we mean "best" in the sense of {!val:score}. *)
```
Comment (gabrielhdt) [inline src/core/tree.ml:358]: line break

File: src/core/unif_rule.ml [hunk 5]
```diff
@@ -23,9 +23,9 @@ let sign : Sign.t =
   Sign.loaded := Files.PathMap.add path s !(Sign.loaded);
   s
 
-(** Symbol representing an atomic unification problem. The term [equiv t
-    u] represents [t ≡ u]. The left-hand side of a unification rule is
-    made of only one unification. *)
+(** Symbol representing an atomic unification problem.
```
Comment (gabrielhdt) [inline src/core/unif_rule.ml:26]: Line break

File: src/core/xtc.ml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 (** This module provides a function to translate a signature to the XTC format
     used in the termination competition.
-    @see <http://cl2-informatik.uibk.ac.at/mercurial.cgi/TPDB/file/tip/xml/xtc.xsd>
+    @see:
+    <http://cl2-informatik.uibk.ac.at/mercurial.cgi/TPDB/file/tip/xml/xtc.xsd>
```
Comment (gabrielhdt) [inline src/core/xtc.ml:4]: I think @fblanqui removed the line break on purpose for ocamldoc formatting.

**Comment by amelieled** [comment]
Maybe I should have put [WIP] in the title. Indeed, there are several points that need to be discussed: they are highlighted by tags like @PROBLEM.

File: src/core/xtc.ml (hunk 6 above)
Comment (fblanqui) [reply src/core/xtc.ml:4]: It's for sanity check.

Comment (fblanqui) [reply src/core/xtc.ml:4]: Yes, you are right @gabrielhdt ;-)

Comment (amelieled) [reply src/core/xtc.ml:4]: OK it's just because the line is too long.

File: src/core/env.ml [hunk 7]
```diff
@@ -37,7 +38,7 @@ let to_prod_box : env -> tbox -> tbox = fun env t ->
   in
   List.fold_left fn t env
 
-(** [to_prod] is an “unboxed” version of [to_prod_box]. *)
+(** [to_prod env t] is an "unboxed" version of [to_prod_box env t]. *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/env.ml:41]: How?

File: src/core/builtin.ml [hunk 8]
```diff
@@ -19,27 +20,29 @@ let htbl : (string, sig_state -> popt -> sym -> unit) Hashtbl.t =
...
+    builtin symbol in scope. It is expected to raise the [Fatal] exception
+    to signal an error.
+    Note that this function should not be called using a [name] for which a
+    check has already been registered. *)
 let register : string -> (sig_state -> popt -> sym -> unit) -> unit =
   fun name check ->
   if Hashtbl.mem htbl name then assert false;
   Hashtbl.add htbl name check
 
+(* @PROBLEM *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/builtin.ml:42]: Here, there are 3 arguments in the documentation, but 4 in reality.
I can't fix this problem.

File: src/core/files.ml [hunk 9]
```diff
@@ -17,15 +17,16 @@ module Path =
     (** Short synonym of [module_path]. *)
     type t = module_path
 
-    (** [compare] is a standard comparing function for module paths. *)
+    (** [compare t1 t2] is a standard comparing function for module paths. *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/files.ml:20]: OK but there is another similary case elsewhere.

File: src/core/env.ml (hunk 7 above)
[the remark this answers is not in the document]
Comment (gabrielhdt) [reply src/core/env.ml]: When activated, the mode automatically replaces quotes

File: src/core/builtin.ml (hunk 8 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/builtin.ml:42]: Please write `[register_expected_type eq pp name fn] calls [register] with a function [check] such that [check ss pos sym] checks that the type of [sym] is equal to some type built with [fn]. It uses [pp] for printing error messages.`

File: src/core/files.ml (hunk 9 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/files.ml]: And the comment is talking about the function, not its result when applied to some arguments.

File: src/core/console.ml [hunk 10]
```diff
@@ -227,7 +227,7 @@ let push_state : unit -> unit = fun () ->
   in
   saved_state := (verbose, loggers, flags) :: !saved_state
 
-(** [pop_state ()] restores the setting saved with [push_stack], removing them
+(** [pop_state ()] restores the setting saved with [push_stack], removing them - @puhs_state ?
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/console.ml:231]: It's the correct function?

File: src/core/extra.ml [hunk 11]
```diff
@@ -197,7 +198,7 @@ module List =
       | x :: xs -> let xs = remove_phys_dups xs in
                    if List.memq x xs then xs else x :: xs
 
-    (** [deconstruct l i] returns a triple [(left_rev, e, right)] where [e] is
+    (** [deconstruct l i] returns a triple [(left_rev, e, right)] where [e] is - PROBLEM
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/extra.ml:201]: The name of this function isn't the same in the ocamldoc.

File: src/core/files.ml [hunk 12]
```diff
@@ -240,7 +241,7 @@ let obj_extension : string = ".lpo"
 (** [legacy_src_extension] is the extension for legacy source files. *)
 let legacy_src_extension : string = ".dk"
 
-(** [file_to_module path] computes the module path that corresponds to [path].
+(** [file_to_module path] computes the module path that corresponds to [path]. - @PROBLEM
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/files.ml:244]: The signature doesn't match.

File: src/core/hrs.ml [hunk 13]
```diff
@@ -13,7 +13,7 @@ let print_sym : sym pp = fun oc s ->
   let print_path = List.pp Format.pp_print_string "_" in
   Format.fprintf oc "c_%a_%s" print_path s.sym_path s.sym_name
 
-(** [print_patt oc p] outputs TPDB format corresponding to the pattern [p], to
+(** [print_patt oc p] outputs TPDB format corresponding to the pattern [p], to - @PROBLEM
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/hrs.ml:16]: The signature doesn't match.

File: src/core/infer.ml [hunk 14]
```diff
@@ -10,14 +10,14 @@ open Print
 let log_infr = new_logger 'i' "infr" "type inference/checking"
 let log_infr = log_infr.logger
 
-(** [destruct_prod n prod] returns a tuple [(env, a)] where [a] is constructed
+(** [destruct_prod n prod] returns a tuple [(env, a)] where [a] is constructed - @PROBLEM
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/infer.ml:13]: The signature doesn't match.

File: src/core/sig_state.ml [hunk 15]
```diff
@@ -52,21 +52,21 @@ type sig_state =
...
-   [s] if [s] is mapped to [name] in [map]. *)
+    [s] if [s] is mapped to [name] in [map]. *)
 let remove_pp_hint :
       sym StrMap.t -> string -> pp_hint SymMap.t -> pp_hint SymMap.t =
   fun map name pp_hints ->
   try SymMap.remove (StrMap.find name map) pp_hints
   with Not_found -> pp_hints
 
-(** [remove_pp_hint_eq map name h pp_hints] removes from [pp_hints] the
+(** [remove_pp_hint_eq map name h pp_hints] removes from [pp_hints] the - @PROBLEM
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/sig_state.ml:69]: I'm not sure that the signature doesn't match.

File: src/core/terms.ml (hunk 2 above)
Comment (amelieled) [reply src/core/terms.ml:402]: It's a bit confusing with the comment in the type `term`

File: src/core/tree.ml [hunk 16]
```diff
@@ -142,12 +142,12 @@ module CP = struct
...
       { pool with variables = IntMap.add i slot pool.variables }
 
-  (** [register_fv slot xs pool] registers a free variables constraint for the
+  (** [register_fv slot xs pool] registers a free variables constraint for the - @PROBLEM
       variables in [xs] on the slot [slot] of the [vars] array in [pool]. *)
   let register_fv : int -> int array -> t -> t = fun i vs pool ->
     { pool with fv_conds = IntMap.add i vs pool.fv_conds }
 
-  (** [constrained_nl i pool] tells whether index [i] in the RHS's environment
+  (** [constrained_nl i pool] tells whether index [i] in the RHS's environment - @PROBLEM
```
Comment (amelieled) [inline src/core/tree.ml:150]: The signature doesn't match.

File: src/core/tree.ml [hunk 17]
```diff
@@ -459,7 +460,7 @@ module CM = struct
     let tc_fst_cmp (tca, _) (tcb, _) = TC.compare tca tcb in
     List.sort_uniq tc_fst_cmp (List.filter_map keep_treecons telst)
 
-  (** [store m c] is true iff a term of column [c] of matrix [m] contains a
+  (** [store m c] is true iff a term of column [c] of matrix [m] contains a - @PROBLEM?
```
Comment (amelieled) [inline src/core/tree.ml:463]: The signature doesn't match.

File: src/core/xtc.ml [hunk 18]
```diff
@@ -28,7 +28,7 @@ let status : sym -> symb_status = fun s ->
     | _         -> Object_level
   in is_arrow_kind !(s.sym_type) false
 
-(** [print_term oc p] outputs XTC format corresponding to the term [t], to
+(** [print_term oc p] outputs XTC format corresponding to the term [t], to - @PROBLEM?
```
Comment (amelieled) [inline src/core/xtc.ml:31]: The signature doesn't match.

File: src/core/xtc.ml [hunk 19]
```diff
@@ -92,7 +92,7 @@ and print_type : int -> string -> term pp = fun i s oc t ->
          (print_type i s) a (print_type i s) b
   | LLet(_,t,u)             -> print_type i s oc (Bindlib.subst u t)
 
-(** [print_rule oc s r] outputs the rule declaration corresponding [r] (on the
+(** [print_rule oc s r] outputs the rule declaration corresponding [r] (on the - PROBLEM?
```
Comment (amelieled) [inline src/core/xtc.ml:95]: The signature doesn't match.

File: src/core/console.ml (hunk 10 above)
[the remark this answers is not in the document]
Comment (gabrielhdt) [reply src/core/console.ml:231]: According to the description, it seems!

File: src/core/env.ml (hunk 7 above)
[the remark this answers is not in the document]
Comment (gabrielhdt) [reply src/core/env.ml]: But don't bother to replace everything, if we want to enforce consistency, we should run a script that makes the replacement.

File: src/core/infer.ml (hunk 14 above)
[the remark this answers is not in the document]
Comment (gabrielhdt) [reply src/core/infer.ml:13]: Are you sure? It seems fine to me.

File: src/core/tree.ml (hunk 16 above)
Comment (gabrielhdt) [reply src/core/tree.ml:150]: I'm not sure to understand what you mean: the signature is `int -> t -> bool` and the documentation applies it to an int `i` and a pool of type `t`, so it looks good to me!

File: src/core/infer.ml (hunk 14 above)
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/infer.ml:13]: My bad...

File: src/core/tree.ml (hunk 16 above)
Comment (amelieled) [reply src/core/tree.ml:150]: It's a mistake too.

File: src/core/files.ml (hunk 12 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/files.ml:244]: Here, you should replace path by fname

File: src/core/hrs.ml (hunk 13 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/hrs.ml:16]: Thanks to fix this kind of problems directly in your PR (print_patt -> print_term).

File: src/core/rewrite.ml [hunk 20]
```diff
@@ -302,7 +302,7 @@ let swap : eq_config -> term -> term -> term -> term -> term =
   let refl_a_l = add_args (Symb cfg.symb_refl) [a; l] in
   add_args (Symb cfg.symb_eqind) [a; r; l; t; pred; refl_a_l]
 
-(** [rewrite ps b po t] rewrites according to the equality proved by [t] in
+(** [rewrite ps b po t] rewrites according to the equality proved by [t] in - @PROBLEM
```
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/rewrite.ml:305]:
```
(** [rewrite ss pos ps l2r p t] rewrites the focused goal of [ps] with the equality proved by [t] in the subterm described by the SSReflect pattern [p]. The term [t] should have a type corresponding to
   an equality. Every occurrence of the first instance of the left-hand side
   is replaced by the right-hand side of the obtained proof (or the reverse if
   [l2r] is false). *)
```

File: src/core/sig_state.ml (hunk 15 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/sig_state.ml:69]:
```
(** [remove_pp_hint map name pp_hints] removes from [pp_hints] the mapping for
   [s] if [name] is mapped to [s] in [map]. *)
let remove_pp_hint :
      sym StrMap.t -> string -> pp_hint SymMap.t -> pp_hint SymMap.t =
  fun map name pp_hints ->
  try SymMap.remove (StrMap.find name map) pp_hints
  with Not_found -> pp_hints

(** [remove_pp_hint_eq map name h pp_hints] removes from [pp_hints] the
   mapping for [s] if [name] is mapped to [(s,h')] in [map], and [eq_pp_hint h
   h' = true]. *)
let remove_pp_hint_eq :
      (sym * Pos.popt) StrMap.t -> string -> pp_hint -> pp_hint SymMap.t
      -> pp_hint SymMap.t =
  fun in_scope name h pp_hints ->
```

File: src/core/sig_state.ml [hunk 21]
```diff
@@ -200,14 +200,14 @@ let open_sign : sig_state -> Sign.t -> sig_state = fun ss sign ->
   in
   {ss with in_scope; builtins; unops; binops; pp_hints}
 
-(** Dummy [sig_state] made from the dummy signature. *)
+(** [dummy : sig_state] made from the dummy signature. *)
```
Comment (fblanqui) [inline src/core/sig_state.ml:203]: no need to put the type in the comment

File: src/core/sign.ml [hunk 22]
```diff
@@ -278,8 +280,8 @@ let read : string -> t = fun fname ->
   in
   reset_timed_refs sign
 
-(* NOTE here, we rely on the fact that a marshaled closure can only be read by
-   processes running the same binary as the one that produced it. *)
+(** NOTE here, we rely on the fact that a marshaled closure can only be read
```
Comment (fblanqui) [inline src/core/sign.ml:283]: keep as normal comment (does not need to appear in the ocaml doc)

File: src/core/sig_state.ml (hunk 21 above)
Comment (gabrielhdt) [reply src/core/sig_state.ml:203]: In general, I don't think we need to follow the pattern `[fun args]` with constants, we just have to explain what they are, hence the first version of the comment.

File: src/core/terms.ml [hunk 23]
```diff
@@ -377,7 +377,7 @@ let _Impl : tbox -> tbox -> tbox =
 let _Abst : tbox -> tbinder Bindlib.box -> tbox =
   Bindlib.box_apply2 (fun a t -> Abst(a,t))
 
-(** [_Meta m ar] lifts the metavariable [m] to the {!type:tbox} type given its
+(** [_Meta u ar] lifts the metavariable [u] to the {!type:tbox} type given its
```
Comment (fblanqui) [inline src/core/terms.ml:380]: Please replace u by m in the code instead.

Comment (fblanqui) [reply src/core/terms.ml:380]: I propose to replace ar by ts as well.

File: src/core/terms.ml (hunk 2 above)
Comment (fblanqui) [reply src/core/terms.ml:402]: `(** [_LLet t a u] lifts a let node to the {!type:tbox} type, given a
    boxed term [t] for the definition, a boxed term [a] for the type, and a boxed binder [u]. *)`

Comment (fblanqui) [reply src/core/terms.ml:402]: Btw, there is a comment missing for _Impl: `(** [Impl a b] lifts a non-dependent product node to the {!type:tbox} type, given
    a boxed term [a] for the domain of the product, and a boxed term [b] for
    its codomain. *)`

File: src/core/xtc.ml (hunk 18 above)
Comment (fblanqui) [reply src/core/xtc.ml:31]: Hello @GuillaumeGen . Could you please explain what is this int argument everywhere? It looks really strange. Would it be possible to remove side effects? And there seems to be a bug in https://github.com/Deducteam/lambdapi/blob/82dccc5074bd42574d8d21b9b39527052d2014cd/src/core/xtc.ml#L49-L50 as what is output is i and not j...

File: src/core/files.ml (hunk 12 above)
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/files.ml:244]: Done + add @raise

File: src/core/hrs.ml (hunk 13 above)
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/hrs.ml:16]: But the signature is `print_term : bool -> term pp`.
The documentation will be not correct if I just replaced print_patt by print_term.

File: src/core/print.ml [hunk 24]
```diff
@@ -253,12 +253,12 @@ let pp_ctxt : ctxt pp = fun oc ctx ->
...
     Format.fprintf oc "%a ⊢ " pp_ctxt ctx
 
-(** [pp_typing oc (c,t,u)] prints the typing constraint [c] to the
+(** [pp_typing oc (ctx,t,u)] prints the typing constraint [ctx] to the
     output channel [oc]. *)
 let pp_typing : constr pp = fun oc (ctx, t, u) ->
   Format.fprintf oc "%a%a : %a" pp_ctxt ctx pp_term t pp_term u
 
-(** [pp_constr oc c] prints the unification constraints [c] to the
+(** [pp_constr oc (ctx,t,u)] prints the unification constraints [ctx] to the
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/print.ml:261]: Done

File: src/core/sig_state.ml (hunk 15 above)
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/sig_state.ml:69]: Done

File: src/core/sig_state.ml (hunk 21 above)
Comment (amelieled) [reply src/core/sig_state.ml:203]: Done

File: src/core/sign.ml (hunk 22 above)
Comment (amelieled) [reply src/core/sign.ml:283]: Done

File: src/core/terms.ml (hunk 23 above)
Comment (amelieled) [reply src/core/terms.ml:380]: Done

File: src/core/terms.ml (hunk 2 above)
Comment (amelieled) [reply src/core/terms.ml:402]: Done

File: src/core/tree.ml [hunk 25]
```diff
@@ -142,12 +142,12 @@ module CP = struct
       (* First occurence of [i], register the slot as a point of reference. *)
       { pool with variables = IntMap.add i slot pool.variables }
 
-  (** [register_fv slot xs pool] registers a free variables constraint for the
+  (** [register_fv slot xs pool] registers a free variables constraint for the - @PROBLEM
```
Comment (amelieled) [inline src/core/tree.ml:145]: Problem

File: src/core/tree.ml [hunk 26]
```diff
@@ -475,7 +476,7 @@ module CM = struct
   let index_var : int VarMap.t -> term -> int = fun vi t ->
     VarMap.find (to_tvar t) vi
 
-  (** [update_aux col slot clause] updates the fields the condition pool and
+  (** [update_aux col slot clause] updates the fields the condition pool and - @PROBLEM?
```
Comment (amelieled) [inline src/core/tree.ml:479]: Problem?

File: src/core/tree.ml (hunk 25 above)
Comment (gabrielhdt) [reply src/core/tree.ml:145]: I don't see any problem.

File: src/core/tree.ml (hunk 17 above)
Comment (gabrielhdt) [reply src/core/tree.ml:463]: Are you sure?

File: src/core/unif.ml [hunk 27]
```diff
@@ -36,17 +36,17 @@ let pp_problem : problem pp = fun oc p ->
...
     environment can hence refer to other variables of the environment whenever
     this is necessary. Note that the produced environment is equivalent to the
-    environment [fst (destruct_prod (Array,length xs) prod)] if the variables
-    of its domain are substituted by those of [xs]. Intuitively,  if [prod] is
-    of the form [∀ (y1:a1) ⋯ (yn:an), a]  then the environment returned by the
-    function is (roughly) [(xn, an{y1≔x1, ⋯, yn-1≔xn-1}) ; ⋯ ; (x1, a1)]. Note
-    that the function raises [Invalid_argument] if [prod] does not evaluate to
-    a sequence of [Array.length xs] dependent products. *)
+    environment [fst (destruct_prod (Array,length xs) t)] if the variables
+    of its domain are substituted by those of [xs]. Intuitively,  if [t] is of
```
Comment (gabrielhdt) [inline src/core/unif.ml:45]: You can see that there are two spaces after the comma (for justification purposes). I think running `M-x fill-paragraph` should normalise that.

File: src/core/sr.ml [hunk 28]
```diff
@@ -123,9 +123,9 @@ let symb_to_tenv
   in
   symb_to_tenv t
 
-(** [check_rule r] checks whether the pre-rule [r] is well-typed in
-   signature state [ss] and then construct the corresponding rule. Note that
-   [Fatal] is raised in case of error. *)
+(** [check_rule pr] checks whether the pre-rule [pr] is well-typed in
+    signature state [ss] and then construct the corresponding rule.
+    Note that [Fatal] is raised in case of error. *)
```
Comment (gabrielhdt) [inline src/core/sr.ml:128]: No need to break line

File: src/core/sign.ml [hunk 29]
```diff
@@ -185,12 +186,12 @@ let unlink : t -> unit = fun sign ->
...
-(** [add_symbol sign ?sym_expo mode name a impl] creates a fresh symbol with
-    name [name] (which should not already be used in [sign]) and with the type
-    [a], in the signature [sign]. The exposition is
-    {!constructor:Terms.sym_exposition.Public} by default, unless [?sym_expo]
-    is precised. The list [impl] tells which arguments is implicit. The
-    created symbol is returned. *)
+(** [add_symbol sign sym_expo sym_prop s a impl] creates a fresh symbol with
+    name [s.elt] (which should not already be used in [sign]) and with the
+    type [a], in the signature [sign].
+    The exposition is [sym_expo] and the property is [sym_prop].
```
Comment (gabrielhdt) [inline src/core/sign.ml:192]: Line break

File: src/core/tree.ml (hunk 17 above)
Comment (amelieled) [reply src/core/tree.ml:463]: What is [vars] array?

File: src/core/tree.ml (hunk 25 above)
Comment (amelieled) [reply src/core/tree.ml:145]: What's [vars] array?

File: src/core/hrs.ml (hunk 13 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/hrs.ml]: Right. `(** [print_term b oc t] outputs to the channel [oc] the term [t] in the HRS format. If [b=true] then type annotations in abstractions are replaced by the variable "m_typ". *)`

[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/hrs.ml]: Btw, this behavior is wrong. It artificially creates non-linear rules as all type annotations are replaced by the *same* variable.

File: src/core/extra.ml [hunk 30]
```diff
@@ -297,8 +299,8 @@ module Array =
       let aux (el, er) (accl, accr) = (el :: accl, er :: accr) in
       Array.fold_right aux a ([], [])
 
-    (** [drop n a] discards the first [n] elements of [a].  The empty array is
-        returned if [n > length a]. *)
+    (** [drop n a] discards the first [n] elements of [a].
+        The empty array is returned if [n > length a]. *)
```
Comment (gabrielhdt) [inline src/core/extra.ml:303]: Line break

File: src/core/hrs.ml (hunk 13 above)
[the remark this answers is not in the document]
Comment (fblanqui) [reply src/core/hrs.ml]: Issue #413

File: src/core/extra.ml (hunk 11 above)
[the remark this answers is not in the document]
Comment (gabrielhdt) [reply src/core/extra.ml:201]: Ah yes, you can replace the name in the documentation.

File: src/core/rewrite.ml (hunk 20 above)
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/rewrite.ml:305]: Done

File: src/core/sr.ml (hunk 28 above)
Comment (amelieled) [reply src/core/sr.ml:128]: You mean: add @raise?

File: src/core/sign.ml (hunk 29 above)
Comment (amelieled) [reply src/core/sign.ml:192]: Done

File: src/core/sign.ml [hunk 31]
```diff
@@ -48,9 +49,9 @@ let mem : t -> string -> bool =
     making copies of terms when loading an object file. *)
 let loaded : t PathMap.t ref = ref PathMap.empty
 
-(* NOTE that the current module is stored in [loaded] so that the symbols that
-   it contains can be qualified with the name of the module. This behavior was
-   inherited from previous versions of Dedukti. *)
+(* NOTE that the current module is stored in [loaded] so that the symbols
+   that it contains can be qualified with the name of the module.
+   This behavior was inherited from previous versions of Dedukti. *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/sign.ml:54]: done

File: src/core/sign.ml [hunk 32]
```diff
@@ -20,13 +20,14 @@ type t =
...
+(** NOTE the [deps] field contains a hashtable binding the [module_path] of
+    the external modules on which the current signature depends to an
+    association list. This association list then maps definable symbols of
+    the external module to additional reduction rules defined in the current
+    signature. *)
 
-(** The empty signature. It's a thunk to force the creation of a new record on
-    each call (and avoid unwanted sharing). *)
+(** [dummy ()] The empty signature. It's a thunk to force the creation of
+    a new record on each call (and avoid unwanted sharing). *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/sign.ml:30]: Done

File: src/core/rewrite.ml [hunk 33]
```diff
@@ -12,14 +12,14 @@ open Print
...
+(** [eq ctx a b] tests the equality of [a] and [b] (up to α-equivalence).
+    It fails if [a] or [b] contain terms of the form [Patt(i,s,e)] or
+    [TEnv(te,env)].  In the process, subterms of the form [TRef(r)] in [a] and
+    [b] may be set with the corresponding value to enforce equality, and
+    variables appearing in [ctx] can be unfolded. In other words, [eq a b] can
     be used to implement non-linear matching. When the
     matching feature is used, one should make sure that [TRef] constructors do
-    not appear both in [t] and in [u] at the same time. Indeed, the references
+    not appear both in [a] and in [b] at the same time. Indeed, the references
     are set naively, without occurrence checking. *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/rewrite.ml:23]: done

File: src/core/tree.ml (hunk 17 above)
Comment (gabrielhdt) [reply src/core/tree.ml:463]: Ah yes, it's a cross reference, it is the array in the `walk_tree` function

File: src/core/extra.ml (hunk 30 above)
Comment (amelieled) [reply src/core/extra.ml:303]: For me the first version is easier to write than this one, but OK.

File: src/core/sr.ml (hunk 28 above)
Comment (gabrielhdt) [reply src/core/sr.ml:128]: Ah yes, I did not mean this, but you're right.

**Comment by gabrielhdt** [comment]
So I have that awk script <https://gist.github.com/gabrielhdt/f2f6b485286b63774a809688d177edfe> that work on many cases but can hurt the caml code in some cases. So I suggest that we first focus on wording and semantics, and I'll adjust cosmetics after.

File: src/core/extra.ml [hunk 34]
```diff
@@ -253,9 +254,10 @@ module Array =
   struct
     include Array
 
-    (** [for_all2 p a1 a2] checks if the corresponding elements of arrays [a1]
-        and [a2] satisfy the predicate [p].  The [Invalid_argument]  exception
-        is raised if the arrays do not have the same size. *)
+    (** [for_all2 f a1 a2] checks if the corresponding elements of arrays [a1]
+        and [a2] satisfy the predicate [f].
+        @raise The [Invalid_argument] exception is raised if the arrays do not
+         have the same size. *)
```
[the remark this answers is not in the document]
Comment (amelieled) [reply src/core/extra.ml]: Done everywhere

**Comment by amelieled** [comment]
Can I delete with PR tomorrow?

**Comment by fblanqui** [comment]
No need to delete it. We can just close it.

**Comment by amelieled** [comment]
I wish they'd been taken into account.

**Comment by fblanqui** [comment]
It may be difficult to merge this PR now.

**Pull request closed without merging**